Examples/Algorithms/Geant4/src/SensitiveSurfaceMapper.cpp

```diff
@@ -0,0 +1,99 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2021 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "ActsExamples/Geant4/SensitiveSurfaceMapper.hpp"
+
+#include "Acts/Definitions/Units.hpp"
+#include "Acts/Geometry/Layer.hpp"
+#include "Acts/Geometry/TrackingGeometry.hpp"
+#include "Acts/Surfaces/Surface.hpp"
+#include "Acts/Utilities/Helpers.hpp"
+
+#include <stdexcept>
+
+#include <G4LogicalVolume.hh>
+#include <G4Material.hh>
+#include <G4VPhysicalVolume.hh>
+#include <globals.hh>
+
+ActsExamples::SensitiveSurfaceMapper::SensitiveSurfaceMapper(
+    const Config& cfg, std::unique_ptr<const Acts::Logger> logger)
+    : m_cfg(cfg), m_logger(std::move(logger)) {
+  if (m_cfg.trackingGeometry == nullptr) {
+    throw std::invalid_argument("No Acts::TrackingGeometry provided.");
+  }
+}
+
+void ActsExamples::SensitiveSurfaceMapper::remapSensitiveNames(
+    G4VPhysicalVolume* g4PhysicalVolume, const Acts::Vector3 motherPosition,
+    int& sCounter) const {
+  auto g4LogicalVolume = g4PhysicalVolume->GetLogicalVolume();
+  auto g4SensitiveDetector = g4LogicalVolume->GetSensitiveDetector();
+
+  G4int nDaughters = g4LogicalVolume->GetNoDaughters();
+
+  constexpr double convertLength = CLHEP::mm / Acts::UnitConstants::mm;
+
+  // Get the relative translation of the G4 object
+  auto g4RelTranslation = g4PhysicalVolume->GetTranslation();
+  Acts::Vector3 g4RelPosition(g4RelTranslation[0] * convertLength,
+                              g4RelTranslation[1] * convertLength,
+                              g4RelTranslation[2] * convertLength);
+
+  if (nDaughters == 0) {
+    std::string volumeMaterialName = g4LogicalVolume->GetMaterial()->GetName();
+    if (g4SensitiveDetector != nullptr or
+        volumeMaterialName.find("Silicon") != std::string::npos) {
```

Comment:
I am not sure of how necessary that would be but maybe we should have in the config a user defined list of material that are considered sensitive instead of just silicon ?

Reply:
We should!

Reply:
This was indeed prepared already (through the Config struct). Implemented now.